File: Lab6/javaScript/main.js
function calcul() {
    let a = parseFloat(document.getElementById("inp1").value); //Зчитування x
    let b = parseFloat(document.getElementById("inp2").value); //Зчитування e
    let n = parseFloat(document.getElementById("inp3").value); //Зчитування e
    let s1 = calculateArea(0, Math.PI, n, func1);
    let s2 = calculateArea(a, b, n, func2);
    document.getElementById("result").innerText = "Result: " + (s1 + s2);
}

function calculateArea(localA, localB, localN, func) {
    let h = (localB - localA) / localN;
    let s = 0;
    for (let i = 1; i <= localN; i++) {
        let xi = localA + i * h - h / 2;
        s += func(xi)
    }
    return s * h
}

function func1(x) {
    return Math.log(2 + Math.sin(x))
}

function func2(x) {
    return Math.pow(Math.atan(x), 2)
}

File: Lab6/golang/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var a, b, n float64
	fmt.Print("a: ")
	_, _ = fmt.Scanln(&a)
	fmt.Print("b: ")
	_, _ = fmt.Scanln(&b)
	fmt.Print("n: ")
	_, _ = fmt.Scanln(&n)
	s1 := calculateArea(0.0, math.Pi, n, func1)
	s2 := calculateArea(a, b, n, func2)
	fmt.Println("Result:", s1+s2)
}
func calculateArea(localA, localB, localN float64, function func(float64) float64) float64 {
	h := (localB - localA) / localN
	var s = 0.0
	for i := 1.0; i <= localN; i++ {
		xi := localA + i*h - h/2
		s += function(xi)
	}
	return s * h
}

func func1(x float64) float64 {
	return math.Log(2 + math.Sin(x))
}

func func2(x float64) float64 {
	return math.Pow(math.Atan(x), 2)
}
